### PR TITLE
:art: reinvent moment.duration

### PR DESCRIPTION
**Pull request by haishengwu-okta**
## Description

- `PasswordReset_spec` and `PasswordExpired_spec` have some tests for the time conversion as well.

## Manual tests

- browsers: https://okta.box.com/s/vggxe2b8vlkb18er5m0qya6aq1dxg728
- outlook: https://okta.box.com/s/ezpri1ebx5ma24va8wyyfbr1vg11ip2n

### Before

```
total 43176
drwxr-xr-x  10 haishengwu  staff      320 Dec 20 15:12 .
drwxr-xr-x   8 haishengwu  staff      256 Dec 20 15:12 ..
-rw-r--r--   1 haishengwu  staff   780355 Dec 20 15:12 okta-sign-in-no-jquery.js
-rw-r--r--   1 haishengwu  staff  5202908 Dec 20 15:12 okta-sign-in-no-jquery.js.map
-rw-r--r--   1 haishengwu  staff  1322495 Dec 20 15:12 okta-sign-in.entry.js
-rw-r--r--   1 haishengwu  staff  2124040 Dec 20 15:12 okta-sign-in.entry.js.map
-rw-r--r--   1 haishengwu  staff  2335659 Dec 20 15:12 okta-sign-in.js
-rw-r--r--   1 haishengwu  staff  3438501 Dec 20 15:12 okta-sign-in.js.map
-rw-r--r--   1 haishengwu  staff   878446 Dec 20 15:12 okta-sign-in.min.js
-rw-r--r--   1 haishengwu  staff  6005841 Dec 20 15:12 okta-sign-in.min.js.map
```

### After

```
total 40832
drwxr-xr-x  10 haishengwu  staff      320 Dec 20 15:09 .
drwxr-xr-x   8 haishengwu  staff      256 Dec 20 15:09 ..
-rw-r--r--   1 haishengwu  staff   740729 Dec 20 15:09 okta-sign-in-no-jquery.js
-rw-r--r--   1 haishengwu  staff  4901807 Dec 20 15:09 okta-sign-in-no-jquery.js.map
-rw-r--r--   1 haishengwu  staff  1204254 Dec 20 15:09 okta-sign-in.entry.js
-rw-r--r--   1 haishengwu  staff  1985528 Dec 20 15:09 okta-sign-in.entry.js.map
-rw-r--r--   1 haishengwu  staff  2217425 Dec 20 15:09 okta-sign-in.js
-rw-r--r--   1 haishengwu  staff  3299992 Dec 20 15:09 okta-sign-in.js.map
-rw-r--r--   1 haishengwu  staff   838820 Dec 20 15:09 okta-sign-in.min.js
-rw-r--r--   1 haishengwu  staff  5704662 Dec 20 15:09 okta-sign-in.min.js.map
```

### CODE REVIEW

File: src/util/TimeUtil.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+    /**
+     * @method getTimeInHighestRelevantUnit
+     * Will return a number in the units of the highest relevant time unit.
+     * Only checks milliseconds, seconds, minutes, hours, and days.
+     * E.g.
+     *   15 minutes -> 15 minutes
+     *   60 minutes ->  1 hours
+     *   90 minutes -> 90 minutes
+     *   24 hours   ->  1 days
```
Comment (mpeng-okta) [inline src/util/TimeUtil.js]: Would this return "1 hours" and "1 days"?

File: packages/@okta/courage-for-signin-widget/webpack.config.js [hunk 2]
```diff
@@ -42,6 +41,13 @@ const webpackConfig = {
       // jsons is from StringUtil
       'vendor/lib/json2': EMPTY,
 
+      // util/handlebars-wrapper -> moment
+      // It's usage for formatting date in handlebar template.
+      // signin-widget doesn't use that feature hence it's safe to comment out.
+      // Could be remove when upgrade courage to latest version from which
+      // we would explicitly import handlerbar plugins in main file.
```
Comment (manueltanzi-okta) [inline packages/@okta/courage-for-signin-widget/webpack.config.js]: nit: "handlebar"

File: src/util/TimeUtil.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
+define(['parse-ms', '@sindresorhus/to-milliseconds'], function (parseMs, toMilliseconds) {
+
+  const MOMENT_UNIT = {
+    MILLISECOND: 'milliseconds',
```
Comment (manueltanzi-okta) [inline src/util/TimeUtil.js]: Do we need to distinguish singulars and plurals? (i.e.: '1 millisecond' vs. ' 2 milliseconds')

File: test/unit/spec/TimeUtil_spec.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+  describe('util/TimeUtil', function () {
+
+    describe('getTimeInHighestRelevantUnit', function () {
+      it('converts millisends', function () {
+        expect(TimeUtil.getTimeInHighestRelevantUnit(0, 'milliseconds')).toEqual({
+          time: 0,
+          unit: 'MILLISECOND',
+        });
+        expect(TimeUtil.getTimeInHighestRelevantUnit(1, 'milliseconds')).toEqual({
```
Comment (manueltanzi-okta) [inline test/unit/spec/TimeUtil_spec.js]: Even if we decide to not handle the edge case of singular (1 millisecond), I think constant name and value should be the same to be less confusing (i.e.: MILLISECONDS = 'milliseconds');

File: src/util/TimeUtil.js (hunk 3 above)
Comment (haishengwu-okta) [reply src/util/TimeUtil.js]: good question!
I wouldn't worry it for now given this is a refactor work and trying to retain identical functionality to the TimeUtil from Courage to mitigate risk.
certainly will revisit once it becomes a problem.

File: src/util/TimeUtil.js (hunk 1 above)
Comment (mpeng-okta) [reply src/util/TimeUtil.js]: Synced up with Haisheng; for the purposes of this PR, it's fine, as it's existing problematic functionality with TimeUtil from Courage

Comment (haishengwu-okta) [reply src/util/TimeUtil.js]: Actually the description is not accurate.
The unit in the return object is canonical.
Updated comments.
